km63/Buts_Oleksandr/laboratory1.sql

```diff
@@ -1,2 +1,132 @@
 -- LABORATORY WORK 1
 -- BY Buts_Oleksandr
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+робити вибірки з таблиць та оновлювати дані
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+create user butts by 1235
+    default tablespace users
+    temporary tablespace temp;
+grant create session to butts;
+grant create any table to butts;
+grant select any table to butts;
+grant update any table to butts;
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+На вулиці стоїть будинок, що має 10 квартир.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+create table house(idh number(2));
+alter table house
+    add constraint idh_k primary key(idh);
+create table flat(idf number(2),idhh number(2));
+alter table flat
+    add constraint idf_k primary key(idf);
+alter table flat
+    add constraint idf_fk foreign key(idhh) references house(idh);
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+grant insert any table to butts;
+grant select any table to butts;
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Який номер замовлення куди входить найдорожчий товар та яке ім'я покупця цього замовлення?
+Виконати завдання в алгебрі Кодда.
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+X=CUSTOMERS WHERE (
+X PROGECT CUSTOMERS. ,CUSTOMERS.CUST_NAME
+
+
+
```

Comment:
SELECT O.ORDER_NUM, CUST_NAME
FROM CUSTOMERS C, ORDERS O, ORDERITEMS OI
WHERE
    OI.ORDER_NUM = O.ORDER_NUM
    and
    O.CUST_ID = C.CUST_ID
    and
    ITEM_PRICE in (
                    SELECT MAX(ITEM_PRICE)
                    FROM ORDERITEMS
    );